File: tools/fix-levels-jsonb.py
#!/usr/bin/env python3
"""
Fix the levels column type mismatch in supabase-data-fixed.sql
Replaces ARRAY[]::text[] with '[]'::jsonb to match the jsonb column type
"""

import os
import re

# Get the path to the SQL file
script_dir = os.path.dirname(os.path.abspath(__file__))
sql_file_path = os.path.join(script_dir, '..', 'supabase-data-fixed.sql')

print('Fixing levels column type mismatch...')

# Read the SQL file
with open(sql_file_path, 'r', encoding='utf-8') as f:
    sql_content = f.read()

print(f'Original file size: {len(sql_content)} bytes')

# Count occurrences before fix
before_count = len(re.findall(r'ARRAY\[\]::text\[\]', sql_content))
print(f'Found {before_count} instances of ARRAY[]::text[]')

# Replace ARRAY[]::text[] with '[]'::jsonb
sql_content = re.sub(r'ARRAY\[\]::text\[\]', "'[]'::jsonb", sql_content)

# Count occurrences after fix
after_count = len(re.findall(r'ARRAY\[\]::text\[\]', sql_content))
print(f'Remaining instances of ARRAY[]::text[]: {after_count}')

# Count new jsonb arrays
jsonb_count = len(re.findall(r"'?\[\]'?::jsonb", sql_content))
print(f'Total instances of []::jsonb: {jsonb_count}')

# Write the fixed content back
with open(sql_file_path, 'w', encoding='utf-8') as f:
    f.write(sql_content)

print(f'\n✓ Fixed! Updated file saved to: {sql_file_path}')
print(f'File size after fix: {len(sql_content)} bytes')

File: tools/fix-levels-jsonb.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Read the SQL file
const sqlFilePath = path.join(__dirname, '..', 'supabase-data-fixed.sql');
let sqlContent = fs.readFileSync(sqlFilePath, 'utf8');

console.log('Fixing levels column type mismatch...');
console.log('Original file size:', sqlContent.length, 'bytes');

// Count occurrences before fix
const beforeCount = (sqlContent.match(/ARRAY\[\]::text\[\]/g) || []).length;
console.log('Found', beforeCount, 'instances of ARRAY[]::text[]');

// Replace ARRAY[]::text[] with '[]'::jsonb
sqlContent = sqlContent.replace(/ARRAY\[\]::text\[\]/g, "'[]'::jsonb");

// Count occurrences after fix
const afterCount = (sqlContent.match(/ARRAY\[\]::text\[\]/g) || []).length;
console.log('Remaining instances of ARRAY[]::text[]:', afterCount);

// Count new jsonb arrays
const jsonbCount = (sqlContent.match(/'?\[\]'?::jsonb/g) || []).length;
console.log('Total instances of []::jsonb:', jsonbCount);

// Write the fixed content back
fs.writeFileSync(sqlFilePath, sqlContent, 'utf8');

console.log('\n✓ Fixed! Updated file saved to:', sqlFilePath);
console.log('File size after fix:', sqlContent.length, 'bytes');
